myia_backend_python/myia_backend_python/python.py

```diff
@@ -441,9 +441,14 @@ def convert_tuple(self, v: tuple, t: AbstractTuple):
         return "()"
 
     def convert_type(self, v, t):
-        myia_type = t.element.xtype()
         # Return type name as a string.
-        return f"'{type_to_np_dtype(myia_type)}'"
+        myia_type = t.element.xtype()
+        if myia_type is None:
+            if isinstance(v, AbstractHandle):
+                return "HandleInstance"
+        if myia_type is Tuple:
+            return "tuple"
+        return f"np.{type_to_np_dtype(myia_type)}"
```

Comment:
I think the code is a bit oddly structured here.

```suggestion
        if isinstance(v, AbstractHandle):
            return "HandleInstance"
        else:
            myia_type = t.element.xtype()
            if myia_type is Tuple:
                return "tuple"
            else:
                return f"np.{type_to_np_dtype(myia_type)}"
```